src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const splittedDate = date.split(separator);
+  const lengthSplitedDate = splittedDate.length;
+  const storeDateObj = {};
+
+  for (let i = 0; i < lengthSplitedDate; i++) {
+    const keyFromFormat = fromFormat[i];
+    const valueFromSplitteDate = splittedDate[i];
+
+    if (keyFromFormat === 'YYYY') {
+      storeDateObj['YY'] = valueFromSplitteDate.slice(2);
+    }
+
+    if (keyFromFormat === 'YY') {
+      if (+valueFromSplitteDate < 30) {
+        storeDateObj['YYYY'] = '20' + valueFromSplitteDate;
+      } else {
+        storeDateObj['YYYY'] = '19' + valueFromSplitteDate;
+      }
+    }
+
+    storeDateObj[keyFromFormat] = valueFromSplitteDate;
+  }
+
+  const separatorOut = toFormat[toFormat.length - 1];
+  const resultDateArr = [];
+  const toFormatDate = toFormat.slice(0, -1);
+
+  for (const i of toFormatDate) {
```

Comment:
that's not index, but a datePart
```suggestion
  for (const datePart of toFormatDate) {
```

Reply:
fixed